File: ReverseLinkedList/recursive.java
public class Solution {
    public ListNode reverseList(ListNode head) {
        return reversedMerge(null, head);
    }

    private ListNode reversedMerge(ListNode target, ListNode head) {
        if (head == null) {
            return target;
        }
        ListNode nextNode = head.next;
        head.next = target;
        return reversedMerge(head, nextNode);
    }
}

File: ReverseLinkedList/recursive.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        return reversed_merge(None, head)

def reversed_merge(target, head):
    if not head:
        return target
    next_ = head.next
    head.next = target
    return reversed_merge(head, next_)
